src/test/skript/tests/syntaxes/sections/EffSecSpawn.sk

```diff
@@ -20,3 +20,55 @@ test "spawn section":
 	# Make sure variables carry over properly
 	assert {_before} is 10 with "value of {_before} should be 10 (got '%{_before}%')"
 	assert {_new var} is 5 with "value of {_new var} should be 5 (got '%{_new var}%')"
+
+test "spawn cats by type" when running minecraft "1.15.2":
+	delete all cats
+	set {_l} to location of spawn of world "world"
+	spawn 5 all black cats at {_l}
+	assert size of all cats = 5 with "Size of all cats is not 5"
+	assert size of all all black cats = 5 with "Size of all all black cats is not 5"
+	spawn 2 siamese cats at {_l}
+	assert size of all cats = 7 with "Size of all cats is not 7"
+	assert size of all siamese cats = 2 with "Size of all siamese cats is not 2"
+	delete all siamese cats
+	assert size of all cats = 5 with "Size of all cats is not 5 after delete 2 siamese cats"
+	delete all cats
+	assert size of all cats = 0 with "Size of all cats is greater than 0 after all were deleted"
+
+test "spawn entities":
+	set {_entities::*} to "allay", "axolotl", "bat", "bee", "blaze", "cat", "cave spider", "chicken" and "cod"
+	add "cow", "creeper", "dolphin", "donkey", "drowned", "elder guardian", "enderman" and "endermite" to {_entities::*}
+	add "evoker", "fox", "frog", "ghast", "glow squid", "goat", "guardian", "hoglin", "horse" and "husk" to {_entities::*}
+	add "iron golem", "llama", "magma cube", "mooshroom", "mule", "ocelot", "panda", "parrot" and "phantom" to {_entities::*}
+	add "pig", "piglin", "piglin brute", "pillager", "polar bear", "pufferfish", "rabbit" and "ravager" to {_entities::*}
+	add "salmon", "sheep", "shulker", "silverfish", "skeleton", "skeleton horse" and "slime" to {_entities::*}
+	add "snow golem", "spider", "squid", "stray", "strider", "tadpole", "trader llama" and "tropical fish" to {_entities::*}
+	add "turtle", "vex", "villager", "vindicator", "wandering trader", "warden", "witch" and "wither skeleton" to {_entities::*}
+	add "wolf", "zoglin", "zombie", "zombie horse", "zombie villager", "zombified piglin" to {_entities::*}
+	add "ender dragon" and "wither" to {_entities::*}
+	add "oak boat", "oak chest boat", "armor stand", "primed tnt", "firework", "arrow", "trident" and "egg" to {_entities::*}
+	add "snowball", "end crystal", "eye of ender", "regular minecart", "falling sand" and "falling gravel" to {_entities::*}
+	add "falling powder snow" to {_entities::*}
+
+	if running minecraft "1.20.0":
+		add "camel" and "sniffer"  to {_entities::*}
+	if running minecraft "1.20.6":
+		add "armadillo" to {_entities::*}
+	if running minecraft "1.21":
+		add "bogged" and "breeze" to {_entities::*}
```

Comment:
can't this all be like `an allay, an axolotl, a bat, a bee, etc`?
then parsing isn't needed

Reply:
I feel like the parsing would be great to keep  as it could combine 2 in 1, not only ensuring that the entity being spawned is of the expected entity type, but also to ensure the parsing works as intended.

Reply:
Well the parsing happens whether you manually parse it or not

Reply:
i guess that is true, you could just add the entity types and then use `parse if` for the version sections

Reply:
I can change it to whichever one is preferred.

Reply:
@sovdeeth What's the conclusion? Keep as is or change to this?